cpp/arcticdb/storage/rocksdb/rocksdb_storage-inl.hpp

```diff
@@ -0,0 +1,170 @@
+/* Copyright 2023 Man Group Operations Limited
+ *
+ * Use of this software is governed by the Business Source License 1.1 included in the file licenses/BSL.txt.
+ *
+ * As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+ */
+
+#include <folly/gen/Base.h>
+#ifndef ARCTICDB_ROCKSDB_STORAGE_H_
+#error "This should only be included by rocksdb_storage.hpp"
+#endif
```

Comment:
Is there a reason for using this pattern?

I think that we must stop using this pattern which is causing linking problem when compiling with `msbuild`.

See https://github.com/man-group/ArcticDB/pull/449#issuecomment-1757567031.

Reply:
Thanks for bringing this to my attention. I've updated the code to follow the new practice defined there.